### PR TITLE
Add subscribe new vote api

### PR DESCRIPTION
**Pull request by Loverush**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: eth/filters/api.go [hunk 1]
```diff
@@ -177,6 +177,40 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Su
 	return rpcSub, nil
 }
 
+// NewVotes creates a subscription that is triggered each time a transaction
+// enters the transaction pool and was signed from one of the transactions this nodes manages.
```
Comment (NashBC) [inline eth/filters/api.go:181]: transaction -> vote

File: eth/filters/api.go [hunk 2]
```diff
@@ -177,6 +177,40 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Su
...
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	rpcSub := notifier.CreateSubscription()
+
+	gopool.Submit(func() {
+		voteHashes := make(chan []common.Hash, 128)
+		voteSub := api.events.SubscribePendingTxs(voteHashes)
```
Comment (NashBC) [inline eth/filters/api.go:192]: SubscribePendingTxs is not right

File: eth/filters/api.go [hunk 3]
```diff
@@ -177,6 +177,40 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Su
...
+func (api *PublicFilterAPI) NewVotes(ctx context.Context) (*rpc.Subscription, error) {
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	rpcSub := notifier.CreateSubscription()
+
+	gopool.Submit(func() {
+		voteHashes := make(chan []common.Hash, 128)
```
Comment (NashBC) [inline eth/filters/api.go:191]: why vote hash not vote itself?

File: eth/filters/api.go [hunk 4]
```diff
@@ -177,6 +177,40 @@ func (api *PublicFilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Su
...
+		voteSub := api.events.SubscribePendingTxs(voteHashes)
+
+		for {
+			select {
+			case hashes := <-voteHashes:
+				// To keep the original behaviour, send a single vote hash in one notification.
+				for _, h := range hashes {
+					notifier.Notify(rpcSub.ID, h)
```
Comment (NashBC) [inline eth/filters/api.go:199]: we should notify vote not vote hash here.

**Review by calmbeing — APPROVED** [review]

File: eth/filters/filter_system_test.go [hunk 5]
```diff
@@ -680,3 +685,73 @@ func flattenLogs(pl [][]*types.Log) []*types.Log {
...
+	var (
+		db      = rawdb.NewMemoryDatabase()
+		backend = &testBackend{db: db}
+		api     = NewPublicFilterAPI(backend, false, deadline, false)
+		votes   = []*types.VoteEnvelope{
+			&types.VoteEnvelope{
+				VoteAddress: types.BLSPublicKey{},
+				Signature:   types.BLSSignature{},
+				Data: &types.VoteData{
+					TargetNumber: uint64(1),
```
Comment (NashBC) [inline eth/filters/filter_system_test.go:701]: Add SourceNumber and Source Hash, otherwise you can't pass CI jobs.

Comment (Loverush) [reply eth/filters/filter_system_test.go:701]: Fixed

**Review by NashBC — APPROVED** [review]

**Pull request merged**